packages/flutter/lib/src/widgets/window.dart

```diff
@@ -2,5 +2,307 @@
 // Use of this source code is governed by a BSD-style license that can be
 // found in the LICENSE file.
 
-/// Placeholder to be used in a future version of Flutter.
-abstract final class Window {}
+import 'dart:ui' show AppExitType, FlutterView;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/services.dart';
+
+import '_window_ffi.dart' if (dart.library.js_util) '_window_web.dart' as window_impl;
+import 'binding.dart';
+import 'framework.dart';
+import 'view.dart';
+
+/// Defines the possible archetypes for a window.
+enum WindowArchetype {
+  /// Defines a traditional window
+  regular,
+}
+
+/// Defines the possible states that a window can be in.
+enum WindowState {
+  /// Window is in its normal state, neither maximized, nor minimized.
+  restored,
+
+  /// Window is maximized, occupying the full screen but still showing the system UI.
+  maximized,
+
+  /// Window is minimized and not visible on the screen.
+  minimized,
+}
+
+/// Defines sizing request for a window.
+class WindowSizing {
+  /// Creates a new [WindowSizing] object.
+  WindowSizing({this.size, this.constraints});
+
+  /// Preferred size of the window. This may not be honored by the platform.
+  final Size? size;
```

Comment:
```suggestion
  /// Preferred size of the window content. This may not be honored by the platform.
  final Size? preferredSize;
```

Reply:
I changed it to preferred size, but `WindowSizing` might be used in future also for window frame on platforms that support it, so I don't think I want to mention window content in the comment.